### ISSUE TITLE
Build warning for modification to SDK file 'agenttypesystem.c'

### ISSUE THREAD

**Issue by funmbi-gd** [Azure/azure-iot-sdk-c#2426]
Hi,

I'd ike to ask about a change made to the SDK file [`agenttypesystem.c`](https://github.com/Azure/azure-iot-sdk-c/blob/main/serializer/src/agenttypesystem.c).
The following lines were added to this file as part of commit [4fcb1446](https://github.com/Azure/azure-iot-sdk-c/commit/4fcb1446c80f5e9f2e9e1b4afaed0e6da9413507#diff-f8fcee387b7f83c84d92115d167333e1cc5eda71d6a76c889872f6a1ee62dbe6)
![agenttypesystem](https://user-images.githubusercontent.com/52742131/212733263-6a633d11-0ef3-43ad-a8ff-dc03a27d7e1e.png)

I'm upgrading the SDK on my device to use a version with these changes and it's producing the following build errors relating to the above modified lines of code:
```
: In function 'CreateAgentDataType_From_String':
sdk/serializer/src/agenttypesystem.c:3497:0: error: ignoring #pragma warning  [-Werror=unknown-pragmas]
 #pragma warning(disable:26451) // warning C26451: overflow in constant arithmetic
 ^
sdk/serializer/src/agenttypesystem.c:3499:0: error: ignoring #pragma warning  [-Werror=unknown-pragmas]
 #pragma warning (default:26451)
 ^
cc1.exe: some warnings being treated as errors
```

Can some context be provided please as to why these lines were added to `agenttypesystem.c`. This will help in determining how to treat this build error on my device. 
Any support is appreciated. Thanks.

**Comment by ericwol-msft** [comment]
This was to suppress warning on VC compile on windows platform. Looks like you have a compile option to issue warning on unknown pragmas. **-Werror=unknown-pragmas** I assume your compiler version does not yet support this pragma.

**Comment by funmbi-gd** [comment]
Thanks for your response. I assume then it should be safe to comment out these lines to help build the code.

**Comment by danewalton** [comment]
Yes we suppress it so if you would like to remove it to successfully build, you may.
